### ISSUE TITLE
Redundant boolean CASE WHEN

### ISSUE THREAD

**Issue by dmateusp** [sqlfluff/sqlfluff#531]
I'm sure you have all seen these as I see them often!

```sql
select
    case
        when n > 0 then true else false end as is_positive
from table_a
```

I would like to add a rule to prefer

```sql
select
    n > 0 as is_positive
from table_a
```

A case when statement is redundant if it uses only one `when` which is turned into `true` or `false`

However, to add a "fix" the rule should also take in account the following:
```sql
select
    case
        when n >= 0 then false else true end as is_negative
from table_a
```

which should be turned into

```sql
select
    case
        not(n >= 0) as is_negative
from table_a
```

**Comment by pwildenhain** [comment]
I actually haven't seen this 😓 but the rule definitely makes sense to me!

**Comment by rbetzler** [comment]
Mind if I give this one a try? I've seen it out in the wild quite a bit lol

**Comment by dmateusp** [comment]
That'd be great @rbetzler thanks :) I've seen your commit and it seems you're already on the right track, but let us know if you need any help

**Comment by rbetzler** [comment]
Could I get a quick look at 0b88499d1320ea8f346a0ff015d7604bac5db556? Given this is my first attempt, I don't want to spam anyone with a bad initial PR lol

If it looks ok, or if it's easier for you all, I can open a PR against upstream. Just let me know what's the preferred process.

That said, I think I've got the basics on (a) returning an error and (b) providing fixes. I ran a few tests on my own company's (Petal) repo and my commit seemed to do the trick.

Hopefully this ain't too bad for a first go 😅

**Comment by dmateusp** [comment]
@rbetzler it's not spam! Just go ahead an open the PR ^^, then we can have a discussion there.

You documented + tested the rule so I'd say it's a 🥇 first go

**Comment by jankatins** [comment]
I also flag this in code reviews, but the logic is actually different if the condition can produce a NULL... Eg when in the initial examples the n is NULL.

**Comment by barrywhart** [comment]
@dmateusp: Can this issue be closed? I see that a PR which mentions it (#798) was merged a few weeks ago.

**Comment by dmateusp** [comment]
Yes! it should have been closed already, closing it now

**Comment by hellomrStampe** [comment]
Hi,

Not sure if this is what resulted in Rule_L043 but it is given me some issues.

I have this where I only want to keep the value of B if the value of A is NULL. If the value of A is not null then B should be NULL

```
select
    case 
        when src.a is null
            then src.b
    end as b_trf,
    src.c
from src
```

However, sqlfluff will try to fix this with "Unnecessary CASE statement. Just use column 'src.a'." resulting in this if I applied the fix

```
select
    src.a as b_trf,
    src.c
from src
```

This is not correct.

@dmateusp if your statement is true "_A case when statement is redundant if it uses only one when_" how would I need to re-write my case statement in order to get the desired result?

Thanks in advance :)
